README.md

```diff
@@ -438,6 +439,73 @@ watcher.each do |line|
 end
 ```
 
+Note that it is not safe to use streaming with Web-servers like this:
+
+```ruby
+class SomeController < ApplicationController
+  def stream_logs
+    response.headers['Content-Type'] = 'text/event-stream'
+
+    watcher = client.watch_pod_log('pod_name', 'namespace')
+    watcher.each do |line|
+      response.stream.write("#{line}\n")
+    end
+  rescue ActionController::Live::ClientDisconnected
+    Rails.logger.info("Streaming logs for cluster##{cluster.id} is done")
+  ensure
+    response.stream.close
+  end
+end
+```
+
+Because watcher reads log with blocking socket, and even if Web-client is disconnected Web-server thread still will be blocked until new logs data arrives, and if new data never arrives you will have resource leaking: one server thread and one socket which is connected to Kubernetes server.
+
+Right way to stream logs with Web-servers:
+
+```ruby
+class SomeController < ApplicationController
+  STREAMING_PING_INTERVAL = 10
+
+  def stream_logs
+    response.headers['Content-Type'] = 'text/event-stream'
+
+    watcher = client.watch_pod_log('pod_name', 'namespace')
+
+    watcher_thread = Thread.new do
+      handle_stream_errors(response, watcher_thread) do
+        watcher.each do |line|
+          response.stream.write("#{line}\n")
+        end
+      end
+    end
+
+    handle_stream_errors(response, watcher_thread) do
+      loop do
+        # Allow watcher thread to execute
+        watcher_thread.join(STREAMING_PING_INTERVAL)
+
+        # Check that conection is alive
+        response.stream.write("ping\n")
+      end
+    end
+  end
+
+  def handle_stream_errors(response, thread, &block)
```

Comment:
They are used both. It's just DRY in error handling. Special thread read logs from Kubernetes, read can be blocked, but thread can't stopped. Web-server worker thread write pings into stream. In case of web-client disconnecting `response.stream.write` will fail and we should clise stream and stop thread to release resource, it can happen in both places.